src/main/scala/viper/gobra/translator/encodings/arrays/ArrayEncoding.scala

```diff
@@ -291,23 +296,23 @@ class ArrayEncoding extends TypeEncoding with SharedArrayEmbedding {
 
   /**
     * Generates:
-    * function arrayConversion(x: [([n]T)@]): ([n]T)°
+    * function arrayConversion(x: [([n]T)@]): (res [n]T)°
     *   requires Footprint[x]
-    *   ensures  [x == result]
+    *   ensures  [x == res]
     * */
-  private val conversionFunc: FunctionGenerator[(BigInt, in.Type)] = new FunctionGenerator[(BigInt, in.Type)]{
-    def genFunction(t: (BigInt, in.Type))(ctx: Context): vpr.Function = {
-      val argType = in.ArrayT(t._1, t._2, Shared)
+  private val conversionFunc: FunctionGenerator[ComponentParameter] = new FunctionGenerator[ComponentParameter]{
+    def genFunction(t: ComponentParameter)(ctx: Context): vpr.Function = {
+      val argType = t.arrayT(Shared)
       val x = in.LocalVar("x", argType)(Source.Parser.Internal)
       val resultType = argType.withAddressability(Exclusive)
       val vResultType = typ(ctx)(resultType)
-      val resultVar = in.LocalVar(Names.freshName, resultType)(Source.Parser.Internal)
+      val resultVar = in.LocalVar("res", resultType)(Source.Parser.Internal)
```

Comment:
Similar to some other comment. How do you know that there is no other variable with the name "res". Is it because there are no other variables? Please write the reason as a comment.